cloud/scope/common.go

```diff
@@ -104,6 +112,47 @@ func CreateLinodeClient(config ClientConfig, opts ...Option) (LinodeClient, erro
 	), nil
 }
 
+func CreateS3Clients(ctx context.Context, crClient K8sClient, cluster infrav1alpha2.LinodeCluster) (S3Client, S3PresignClient, error) {
+	var (
+		configOpts = []func(*awsconfig.LoadOptions) error{
+			awsconfig.WithRegion("auto"),
+		}
+
+		clientOpts = []func(*s3.Options){}
+	)
+
+	// If we have a cluster object store bucket, get its configuration.
+	if cluster.Spec.ObjectStore != nil {
+		secret, err := getCredentials(ctx, crClient, cluster.Spec.ObjectStore.CredentialsRef, cluster.GetNamespace())
+		if err == nil {
```

Comment:
Curious why we're doing this instead of the usual `err != nil`

Reply:
Because these extra configuration options in the AWS SDK clients are only set if the credentials are successfully retrieved.